### ISSUE TITLE
Errors on startup.

### ISSUE THREAD

**Issue by mikebronner** [robb-j/nova-yaml-language-server#4]
Hi, just installed your extension, looks like it has a lot of promise! :)

I just got the following errors when opening a project:

```
YAML[15:49:14.783000] Activated!

YAML[15:49:15.023000] Registering request handler for workspace/didChangeWorkspaceFolders failed.

YAML[15:49:15.025000] (node:53481) UnhandledPromiseRejectionWarning: Error: Invalid parameter: registrations
Registrations must be of type Registration[]
    at handleResponse (/Users/mike/Library/Application Support/Nova/Extensions/robb-j.yaml/dependencyManagement/node_modules/vscode-jsonrpc/lib/main.js:436:48)
    at processMessageQueue (/Users/mike/Library/Application Support/Nova/Extensions/robb-j.yaml/dependencyManagement/node_modules/vscode-jsonrpc/lib/main.js:263:17)
    at Immediate.<anonymous> (/Users/mike/Library/Application Support/Nova/Extensions/robb-j.yaml/dependencyManagement/node_modules/vscode-jsonrpc/lib/main.js:247:13)
    at processImmediate (internal/timers.js:458:21)
(Use `node --trace-warnings ...` to show where the warning was created)

YAML[15:49:15.025000] (node:53481) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

**Comment by robb-j** [comment]
Hi, I had noticed these errors before so I put the note about it in the extension readme. They don't cause any issues from my testing.

It actually looks like it is fixed in version [0.14.0 of the Yaml server](https://github.com/redhat-developer/yaml-language-server/blob/master/CHANGELOG.md#0140), I'll update it soon after #5 is merged.

**Comment by mikebronner** [comment]
Thanks @robb-j ! :)

**Comment by robb-j** [comment]
It looks like upgrading the yaml server didn't fix this. I previously made sure it doesn't crash the server with a `--unhandled-rejections=warn`, which isn't ideal but means it won't kill the node.js process for now. Hopefully panic will implement more of the LSP soon.

I'll leave this issue open for now in case things change in the future.

**Comment by robb-j** [comment]
This is being tracked by panic and there is a devforum post to it below, I'll come back to it once there are any updates from panic / redhat-developers

https://devforum.nova.app/t/lsp-import-action-error/860

**Comment by robb-j** [comment]
I've been doing a bit more digging on this and it seems in the case of Red Hat's yaml server they did fix it. But then they refactored their logic around and moved to [vscode-languageserver](https://github.com/microsoft/vscode-languageserver-node/) code which has the same [bug](https://github.com/microsoft/vscode-languageserver-node/issues/713). From reading through that it seems it has been [fixed](https://github.com/microsoft/vscode-languageserver-node/commit/5bc82ac3c1c656aa70615c9e3e065fa04e41506e) and is scheduled for release in vscode-languageserver version 7.1.0.

So when that is released, and yaml server upgrades to 7.1.0, this might be fixed?

**Comment by robb-j** [comment]
As of nova-yaml 1.4.1, I'm not seeing any errors in the logs any more during startup! The underlying server has changed quite a bit since its own v1 and seems to be more LSP complaint (and nova is too). I'll close this for now, but feel free to re-open if you start seeing messages again.
